controller/array_action/errors.py

```diff
@@ -183,7 +187,7 @@ def __init__(self, connectivity_type):
         self.message = messages.UnsupportedConnectivityTypeError_message.format(connectivity_type)
 
 
-class ExpectedSnapshotButFoundVolumeError(BaseArrayActionException):
+class ExpectedSnapshotButFoundVolumeError(InvalidArgumentError):
```

Comment:
what about `PoolParameterIsMissing` and `IllegalObjectName`?
I know they are common, but I think they are also a type of `InvalidArgumentError`

Reply:
Done